CONTRIBUTING.md

```diff
@@ -0,0 +1,150 @@
+
+Documentation contributions of any size are welcome! Feel free to file a PR even if you're just rewording a sentence to be more clear, or fixing a spelling mistake!
+
+To contribute documentation:
+
+* [Set up the project](#project-setup).
+* Edit or add any relevant documentation.
+* Make sure your changes are formatted correctly and consistently with the rest of the documentation.
+* Re-read what you wrote, and possibly run a spellchecker on it to make sure you didn't miss anything.
+* In your commit message(s), begin the first line with `docs: `. For example: `docs: Adding a doc contrib section to CONTRIBUTING.md`.
+* Write clear, concise commit message(s) using [best practices for commit message formatting](https://github.com/phonegap/phonegap/wiki/Git-Commit-Message-Format).
+* Go to https://github.com/zkat/pacote/pulls and open a new pull request with your changes.
+* If your PR is connected to an open issue, add a line in your PR's description that says `Fixes: #123`, where `#123` is the number of the issue you're fixing.
+
+Once you've filed the PR:
+
+* One or more maintainers will use GitHub's review feature to review your PR.
+* If the maintainer asks for any changes, edit your changes, push, and ask for another review.
+* If the maintainer decides to pass on your PR, they will thank you for the contribution and explain why they won't be accepting the changes. That's ok! We still really appreciate you taking the time to do it, and we don't take that lightly. 💚
+* If your PR gets accepted, it will be marked as such, and merged into the `latest` branch soon after. Your contribution will be distributed to the masses next time the maintainers [tag a release](#tag-a-release)
+
+## Contribute Code
+
+We like code commits a lot! They're super handy, and they keep the project going and doing the work it needs to do to be useful to others.
+
+Code contributions of just about any size are acceptable!
+
+The main difference between code contributions and documentation contributions is that contributing code requires inclusion of relevant tests for the code being added or changed. Contributions without accompanying tests will be held off until a test is added, unless the maintainers consider the specific tests to be either impossible, or way too much of a burden for such a contribution.
+
+To contribute code:
+
+* [Set up the project](#project-setup).
+* Make any necessary changes to the source code.
+* Include any [additional documentation](#contribute-documentation) the changes might need.
+* Write tests that verify that your contribution works as expected.
+* In your commit message(s), begin the first line with `COMPONENT:`, where `COMPONENT` is the relevant component, feature, or section of the code. There's no hard line for this, but look at the commit history for ideas.
```

Comment:
I feel like I should clarify this cause it got confusing -- and as far as bugfixes though, I think I'm more interested in the _component_ the commit affects, rather than whether something is a bugfix or a feature tweak or whatever? So `extract: fix tarball lockup issue` rather than `fix: tarball extract issue`.

An alternative is to just go ahead and use angular-style commit messages, which have a standardized format and some automated tooling built around them. See: https://github.com/conventional-changelog/conventional-changelog-angular/blob/master/convention.md

What do you think? I'm not interested in using semantic-release, but this format would actually let us autogenerate changelogs, which would be handy.